Rubberduck.Parsing/VBA/RubberduckParser.cs

```diff
@@ -613,11 +613,9 @@ private Declaration CreateProjectDeclaration(QualifiedModuleName projectQualifie
 
         private void ResolveReferences(DeclarationFinder finder, VBComponent component, IParseTree tree)
         {
+            Debug.Assert(State.Status == ParserState.ResolvedDeclarations);
+
             var state = _state.GetModuleState(component);
```

Comment:
`var state` should actually be unnecessary for the rest of the method, since the catch blocks should access `_state` instead of setting this variable.